### ISSUE TITLE
Optimize non-float null parquet read and improve user experience

### ISSUE THREAD

**Issue by stress-tess** [Bears-R-Us/arkouda#3095]
some great ideas @bmcdonald3 had for improving #3094! This involves an optimization and improving user experience by not requiring them to set a flag

> Null flag was a good idea! One alternative would be, rather than modifying the error message, we could see if that particular error is thrown and then, if it is, run the nan code automatically. It would read twice in that case though, so I'm not sure what the best decision is.
> 
> Also, I think that we can do some optimizations here, but probably best left for another PR. With strings, we have to worry about data being cleaned up because it is memory being allocated by Arrow, but since all our memory here is allocated in Chapel, we don't have that problem.
> 
> The optimization would to be to read in batches into a third temporary array and then do the assignment in parallel on the Chapel side of things.

_Originally posted by @bmcdonald3 in https://github.com/Bears-R-Us/arkouda/pull/3094#pullrequestreview-1994494036_
